lib/ansible/modules/cloud/azure/azure_rm_keyvault.py

```diff
@@ -380,16 +380,16 @@ def exec_module(self, **kwargs):
                             if n.get('application_id', None) != o.get('application_id', None):
                                 self.to_do = Actions.Update
                                 break
-                            if sorted(n.get('keys', [])) != sorted(o.get('keys', [])):
+                            if sorted(n.get('permissions', {}).get('keys', []) or []) != sorted(o.get('permissions', {}).get('keys', []) or []):
```

Comment:
Is the `or []` necessary?

Reply:
Hey @gavinfish I think I can answer that question as I have tested the output

If you have this:

```json
'permissions': {
                    'keys':
                    None,
                    'secrets': [
                        'get', 'list', 'set', 'delete', 'recover', 'backup',
                        'restore'
                    ],
                    'certificates': ['get', 'list', 'delete', 'create'],
                    'storage':
                    None
                }
```

Where `keys: None`, you will get

```shell
TypeError: 'NoneType' object is not iterable
```

The `[]` fixed that and returns an empty list

Regards

Reply:
Seems like the original code has already having this bug, thanks for clarifying.

Reply:
Thank you!

@gavinfish  Are you going to move this change to collections or needs to be done by @Eugen1987 ?

Guys, tell me if you need help!

Reply:
Feel free to create a PR to Azure collection. All the Azure modules will be maintained  there soon.